src/compiler/checker.ts

```diff
@@ -6662,46 +6665,96 @@ namespace ts {
             return s.valueDeclaration ? s.valueDeclaration.kind : SyntaxKind.PropertyDeclaration;
         }
 
-        function getDeclarationFlagsFromSymbol(s: Symbol) {
+        function getDeclarationFlagsFromSymbol(s: Symbol): NodeFlags {
             return s.valueDeclaration ? getCombinedNodeFlags(s.valueDeclaration) : s.flags & SymbolFlags.Prototype ? NodeFlags.Public | NodeFlags.Static : 0;
         }
 
-        function checkClassPropertyAccess(node: PropertyAccessExpression | QualifiedName, left: Expression | QualifiedName, type: Type, prop: Symbol) {
+        /**
+         * Check whether the requested property access is valid.
+         * Returns true if node is a valid property access, and false otherwise.
+         * @param node The node to be checked.
+         * @param left The left hand side of the property access (e.g.: the super in `super.foo`).
+         * @param type The type of left.
+         * @param prop The symbol for the right hand side of the property access.
+         */
+        function checkClassPropertyAccess(node: PropertyAccessExpression | QualifiedName, left: Expression | QualifiedName, type: Type, prop: Symbol): boolean {

```

Comment:
Well, all the values are precomputed by the (2) callers because they need them to perform additional checks. 

Additionally, `prop` is obtained differently between the two callers. The difference is, effectively: 

1) In `checkPropertyAccessExpressionOrQualifiedName()` we compute
`prop = getPropertyOfType(getApparentType(getWidenedType(type)), right.text)` while

2) In `isValidPropertyAccess()` we compute
` prop = getPropertyOfType(getWidenedType(type), propertyName)`.

I am unsure if this is an issue.